src/main/java/org/jfrog/hudson/CredentialsConfig.java

```diff
@@ -36,7 +36,7 @@
     @DataBoundConstructor
```

Comment:
I suggest modifying the constructor as follows, instead of the proposed change:
```
    public CredentialsConfig(String username, String password, String credentialsId, Boolean overridingCredentials) {
        this.overridingCredentials = overridingCredentials == null ? false : overridingCredentials;
        if (overridingCredentials == null || overridingCredentials.equals(Boolean.TRUE)) {
            this.credentials = new Credentials(username, password);
        }
        this.credentialsId = credentialsId;
    }
```
Basically, taking *credentialsId* out of the if statement. This is because when the plugin is configured to use the Credentials Plugin,  the *overridingCredentials* checkbox is invisible.
Please let me know if the above change also resolves the issue. If so, feel free to commit the code.

Reply:
Updated. It works as expected.
But I am still confused about the value of **overridingCredentials**, overridingCredentials checkbox is invisible while using Credentials Plugin, why the value of overridingCredentials is false ?

Reply:
Let me try to explain:
When the plugin is configured **not** to use the Credentials Plugin (through *Manage --> Configure System --> Artifactory*), the jobs' UI allows overriding the default resolver and deployer credentials with a username and password. This configuration depends on a checkbox, which corresponds to the *overridingCredentials* boolean in the server side code.
When the plugin is configured to use the Credentials Plugin however, this checkbox does not exist in the UI and the *overridingCredentials* is therefore not relevant.